sorc/chgres_cube.fd/grib2_util.F90

```diff
@@ -74,4 +74,171 @@ subroutine convert_omega(omega,p,t,q,clb,cub)
 
 end subroutine convert_omega
 
+function to_upper(strIn) result(strOut)
+! Adapted from http://www.star.le.ac.uk/~cgp/fortran.html (25 May 2012)
+! Original author: Clive Page
+
+     implicit none
+
+     character(len=*), intent(in) :: strIn
+     character(len=len(strIn)) :: strOut
+     integer :: i,j
+
+     do i = 1, len(strIn)
+          j = iachar(strIn(i:i))
+          if (j>= iachar("a") .and. j<=iachar("z") ) then
+               strOut(i:i) = achar(iachar(strIn(i:i))-32)
+          else
+               strOut(i:i) = strIn(i:i)
+          end if
+     end do
+
+end function to_upper
+
+subroutine write_temp(filename,fieldname,localpet,field_input,field_target)
```

Comment:
Is this routine used?

Reply:
No. I wrote that a while back to output data mid-exection for debugging purposes. I'll remove it from this branch.